inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -285,3 +301,47 @@ def refextract(obj, eng):
     elif len(text_references) > len(pdf_references):
         obj.log.info('Extracted %d references from text.', len(text_references))
         obj.data['references'] = text_references
+
+
+def _normalize(s):
+    if not s:
+        return
+
+    result = RE_ALPHANUMERIC.sub(' ', s)
+    result = ' '.join(result.split())
+    result = result.upper()
+
+    if not result:
+        return
+    return result
+
+
+@with_debug_logging
+def normalize_journal_titles(obj, eng):
+    initial_journal_titles = get_value(obj.data, 'publication_info.journal_title')
+
+    refextract_journal_kb_path = current_app.config['REFEXTRACT_JOURNAL_KB_PATH']
+    with codecs.open(refextract_journal_kb_path, mode='r', encoding='utf-8') as f:
+        lines = f.readlines()
+
+        for index, title in enumerate(initial_journal_titles):
+            journal_title = _normalize(title)
+
+            for line in lines:
+                normalized_title_mapping = line.split('---')
+                variant_title = normalized_title_mapping[0]
+                normalized_title = normalized_title_mapping[1]
+
+                if journal_title == variant_title:
+                    journal_title = normalized_title.strip('\n')
+
+                    obj.data['publication_info'][index]['journal_title'] = journal_title
```

Comment:
This index is wrong if there is a `publication_info` with no `journal_title`.

Reply:
This is still wrong, the problem is the following. If the record contains something like
```
{
    'publication_info': [
        {
	    'cnum': 'C01-01-01',
	},
	{
	    'journal_title': 'Journal of Foo',
	    'journal_volume': '42',
	    'artid': 12345,
	}
    ]
}
```
then `enumerate(get_value(obj.data, 'publication_info.journal_title'))` is `[(0, 'Journal of Foo')]`, but `obj.data['publication_info'][0]['journal_title']` does not exist.